### DISCUSSION TITLE
(UserDefinition)Authorization.UserDefinition UserId/Name - How to Get

### DISCUSSION THREAD

**Discussion by coolvijaymca** [serenity-is/Serenity#6719] [Help] [answered]
Guys any idea how to get user name and other user details in this framework.
This is being default application without multi-tenant 
I know this was being used in .NET 5 but in .NET Core I am not able to get it any help will be appreciated. 

-VJ

**Comment by wezmag** [comment, marked as answer]
The easiest way to get UserId is `User.GetIdentifier()`. 

To get other information, you need to
1. Inject `IUserRetrieveService` in your class.
2. use `User.GetUserDefinition<UserDefinition>(UserRetrieveService)` to get `UserDefinition`
3. Once you get UserDefinition, you can access to its properties.

[GetUserDefinition](https://github.com/serenity-is/docs/blob/a89a776a0136966926bbc5df3176d4c4dad86cc7/api/dotnet/Serenity.Net.Core/Serenity/AuthorizationExtensions/GetUserDefinition.md)

If `User` is inaccessible in your calss, you can inject `IUserAccessor` in you class. 
Then use `UserAccessor.User` to get `User`.

**Reply by coolvijaymca** [reply]
Thank You, that was a hint. I was able to get it. Putting code below for community.

`protected IUserRetrieveService UserRetrieveService { get; }
protected IUserAccessor UserAccessor { get; }
  
public MyRowBehavior(IUserRetrieveService userRetrieveService, IUserAccessor userAccessor) // this is your class constructor
    {
        UserAccessor = userAccessor ?? throw new ArgumentNullException(nameof(userAccessor));
        UserRetrieveService = userRetrieveService ?? throw new ArgumentNullException(nameof(userRetrieveService));
    }
public void myFunction(){
        var username = UserAccessor.User?.Identity?.Name;
        var user = UserRetrieveService.ByUsername(username)//Here user contain all information.
}`
